### ISSUE TITLE
top menu remains open when clicking away into iframe

### ISSUE THREAD

**Issue by ivanov** [jupyterlab/jupyterlab#6679]
**Describe the bug**
The top menu remains open when clicking inside an iframe

**To Reproduce**

1. Click on one of the References in the Help menu (like "Jupyter Reference")
2. Click on one of the top level menus again.
3. Click somewhere inside the reference tab that was created.
4. The menu remains visible and does not dissappear when you click away inside the reference tab that was open (and it is likely that this tab takes up the majority of the content space).

**Expected behavior**
Menu should disappear at 3.

- JupyterLab 1.0.0-alpha12

**Additional context**
This is probably due to the lack of interaction between the events in the IFrame and JupyterLab / Phosphor.

**Comment by jasongrout** [comment]
Indeed, phosphor sets up a mousedown handler on the document when the menu is attached, and listens there for any mousedown outside of the menu, in which case it closes the menu:

https://github.com/phosphorjs/phosphor/blob/fa48660e1bc12bf742e752073a6e8804113a921d/packages/widgets/src/menu.ts#L791

Presumably the iframe is swallowing mousedown events. However, the iframe *is* getting focused, so perhaps there should be an additional check for focus changing and the new focused element not including something in the menu? We could perhaps do this outside of jlab (when menu opens, attach a focus handler to the document, when triggered close menu if appropriate, otherwise listen to menu messages or the widget's dispose signal or something.

**Comment by jasongrout** [comment]
@ivanov - we're nearing the planned 1.0 release date. Do you still feel this is a 1.0 issue?
